### PR TITLE
Running pytest in multi-threads

### PR DESCRIPTION
**Pull request by sleepwithcoffee**
*Issue:* #417

*Description of changes:*

Automatically detect available CPU cores and run pytest in multi-threads

### CODE REVIEW

**Review by hawflau — APPROVED** [review]
LGTM! Thanks for the contribution!

File: .github/workflows/build_test_invoke.yml [hunk 1]
```diff
@@ -59,7 +59,7 @@ jobs:
       - name: Run build tests for tests/unit/
         run: |
           pip install -r requirements.txt
-          pytest -vvv tests/unit/
+          pytest -n $(python -c "import psutil; print(psutil.cpu_count(logical=True))") -vvv tests/unit/
```
Comment (mndeveci) [inline .github/workflows/build_test_invoke.yml:62]: Actually, we don't need `psutil` for this, we can just use `-n auto` flag from xdist directly. See https://pytest-xdist.readthedocs.io/en/stable/distribution.html

**Review by mndeveci — CHANGES_REQUESTED** [review]
Thanks for your contribution, added a comment to make it simpler.

File: .github/workflows/build_test_invoke.yml (hunk 1 above)
Comment (sleepwithcoffee) [reply .github/workflows/build_test_invoke.yml:62]: thanks @mndeveci for you suggestion, let me update my PR

Comment (sleepwithcoffee) [reply .github/workflows/build_test_invoke.yml]: updated

**Review by mndeveci — APPROVED** [review]
Thanks for addressing the comments, LGTM!

**Pull request merged**